crates/batcher/src/lib.rs

```diff
@@ -1245,17 +1309,22 @@ impl Batcher {
             return None;
         }
 
-        // Check if a batch is currently being posted
-        let mut batch_posting = self.posting_batch.lock().await;
-        if *batch_posting {
-            info!(
-                "Batch is currently being posted. Waiting for the current batch to be finalized..."
-            );
-            return None;
+        info!("Batch building: started, acquiring lock to stop processing new messages...");
+        let _building_batch_mutex = self.building_batch_mutex.lock().await;
+
+        info!("Batch building: waiting until all the ongoing messages finish");
```

Comment:
info!("Batch building: waiting until all the user messages and proofs get processed");